### DISCUSSION TITLE
Will the PostgreSQL database contain all data from genesis block if indexer runs in archival mode?

### DISCUSSION THREAD

**Discussion by WuTao18** [near/near-indexer-for-explorer#320] [Q&A]
Or indexer will only add the data since sync started to the database, even if it is running in archival mode?

**Comment by khorolets** [comment]
> Or indexer will only add the data since sync started to the database, even if it is running in archival mode?

Indexer adds data to the database starting from the block you've said indexer to start regardless the mode. 

So if you want all the data you need to start an indexer from the genesis block in archival mode and add the key `--store-genesis` (once)
